Training24/Test.c

```diff
@@ -1,19 +1,159 @@
 // -------------------------------------
 // Test.c
-// Program on main branch.
+// Program on A5 branch.
+// Implements test cases and checks the user input for sort & search in integer array.
 // ------------------------------------------------------------------------------------
 #include <stdio.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <conio.h> 
 #include "Header.h"
 
-int main() {
-const char* message = GetMessage ();
-printf ("%s\n", message);
-return;
+#define RED "\x1b[31m"
+#define GREEN "\x1b[32m"
+#define BLUE "\x1b[34m"
+#define MAGENTA "\x1b[35m"
+#define CYAN "\x1b[36m"
+#define RESET "\x1b[0m"
+#define MAX_SIZE 100
+
+/// <summary>Run test for sortiing array.</summary>
+void TestSorting ();
+
+/// <summary>Run test for searching element in sorted array.</summary>
+void TestSearching ();
+
+/// <summary>Gets valid user input.</summary>
+int GetUserInput (const char* input);
+
+/// <summary>Sorts and search for user input.</summary>
+void SortAndSearchUserInput ();
+
+int main () {
+   char choice;
+   while (1) {
+      printf ("\nChoose an option:\n"
+         "1. Run Test Cases\n"
+         "2. Enter User Input\n"
+         "3. Exit\n"
+         "\nEnter your choice (1-3): ");
+      choice = getche ();
+      printf ("\n");
+      switch (choice) {
+      case '1': TestSorting (); TestSearching (); break;
+      case '2': SortAndSearchUserInput (); break;
+      case '3': printf (MAGENTA "\n   EXITING THE PROGRAM.....!!\n" RESET); return 0;
+      default: printf (RED "\nInvalid choice. Please enter 1, 2, or 3.\n" RESET); break;
+      }
+   }
+}
+
+void TestSorting () {
+   int testCases[][MAX_SIZE] = {
+      {1, 2, 3, 4, 5, 6, 7, 8},
+      {0, -3, 1, 3, 5, 7, -9},
+      {10, -20, -30, 40, 0},
+      {3, -3, 3, -3, -3, -3},
+      {5, 5, 5, 5, 5, 5, 5, 5},
+   },
+   arrSizes[] = { 8, 7, 5, 6, 8 };
+   printf (CYAN "\n   ----- TEST CASES FOR SORTING -----\n" RESET);
+   for (int i = 0; i < 5; i++) {
+      int size = arrSizes[i];
+      printf ("\nUnsorted Array: [ ");
+      for (int j = 0; j < size; j++) printf ("%d ", testCases[i][j]);
+      printf ("]\n");
+      BubbleSort (testCases[i], size);
+      printf ("Sorted Array: [ ");
+      for (int j = 0; j < size; j++) printf ("%d ", testCases[i][j]);
+      printf ("]\n");
+      bool isSorted = true;
+      for (int k = 0; k < size - 1; k++) {
+         if (testCases[i][k] > testCases[i][k + 1]) {
+            isSorted = false;
+            break;
+         }
+      }
+      printf ("Result: %s\n", isSorted ? GREEN"PASS"RESET : RED"FAIL"RESET);
+   }
+}
+
+void TestSearching () {
+   int testCases[][MAX_SIZE] = {
+       {1, 2, 3, 4, 5, 6, 7, 8},
+       {0, -3, 1, 3, 5, 7, -9},
+       {10, -20, -30, 40, 0},
+       {3, -3, 3, -3, -3, -3},
+       {5, 5, 5, 5, 5, 5, 5, 5}
+   },
+      arrSizes[] = { 8, 7, 5, 6, 8 },
+      expectedIndex[] = { 5, 0, 1, 2, 3 },
+      searchElements[] = { 6, -9, -20, -3, 5 };
```

Comment:
The problem statement was to find the first occurrence of the element in the array.